magento/pwa-studio
# CIF-2826: support setting full graphql endpoint in useAdapter

## Description

This changes allows users of Peregrine `useAdapter()` to specify the full graphql endpoint to use. 

In the context of AEM the Commerce Integration Framework uses Peregrine based components to accelerate development of transactional frontend components such as Cart, Checkout etc. In this setup a configuration maintained in AEM defines the endpoint the graphql server is running on. This endpoint is different from the default `<origin>/graphql` as it is used for "pure" Adobe Commerce deployments. Customers can even freely define the endpoint. 

In the default setup a reverse proxy is used to route the graphql traffic to the Adobe Commerce backend. This proxy is per default listening on `/api/graphql`. However, as said above customers could use any path, or even an absolute URL to an API gateway for example.

In the past we only supported Peregrine 11.x without using `useAdapter()` as we set up the context for peregrine manually (ApolloClient, Redux). This is not possible with 12.x anymore due to [`clearCacheData()`](https://github.com/magento/pwa-studio/blob/v12.3.0/packages/peregrine/lib/talons/Adapter/useAdapter.js#L272) being added to the apollo client instance, which is for example used in [SignIn](https://github.com/magento/pwa-studio/blob/v12.3.0/packages/peregrine/lib/talons/SignIn/useSignIn.js#L85-L86). The above described change is necessary for use to make use of `useAdapter()` and so pass those internal function references down to the components. 

## Related Issue

CIF-2826

## Verification Steps

The change is minor, full unit test coverage is provided and should be sufficient to verify this change. 

If manual verification is needed please ping me.

## Checklist
